ext-apiserver/pkg/apis/cluster/v1alpha1/cluster_types_test.go

```diff
@@ -15,65 +14,68 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-
 package v1alpha1_test
 
 import (
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	. "k8s.io/kube-deploy/ext-apiserver/pkg/apis/cluster/v1alpha1"
 	. "k8s.io/kube-deploy/ext-apiserver/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
+	"reflect"
+	"testing"
+	"k8s.io/kube-deploy/ext-apiserver/pkg/client/clientset_generated/clientset"
 )
 
-var _ = Describe("Cluster", func() {
+func crudAccessToClusterClient(t *testing.T, cs *clientset.Clientset) {
 	var instance Cluster
 	var expected Cluster
 	var client ClusterInterface
 
-	BeforeEach(func() {
-		instance = Cluster{}
-		instance.Name = "instance-1"
-
-		expected = instance
-	})
-
-	AfterEach(func() {
-		client.Delete(instance.Name, &metav1.DeleteOptions{})
-	})
-
-	Describe("when sending a storage request", func() {
-		Context("for a valid config", func() {
-			It("should provide CRUD access to the object", func() {
-				client = cs.ClusterV1alpha1().Clusters("cluster-test-valid")
-
-				By("returning success from the create request")
-				actual, err := client.Create(&instance)
-				Expect(err).ShouldNot(HaveOccurred())
-
-				By("defaulting the expected fields")
-				Expect(actual.Spec).To(Equal(expected.Spec))
-
-				By("returning the item for list requests")
-				result, err := client.List(metav1.ListOptions{})
-				Expect(err).ShouldNot(HaveOccurred())
-				Expect(result.Items).To(HaveLen(1))
-				Expect(result.Items[0].Spec).To(Equal(expected.Spec))
-
-				By("returning the item for get requests")
-				actual, err = client.Get(instance.Name, metav1.GetOptions{})
-				Expect(err).ShouldNot(HaveOccurred())
-				Expect(actual.Spec).To(Equal(expected.Spec))
-
-				By("deleting the item for delete requests")
-				err = client.Delete(instance.Name, &metav1.DeleteOptions{})
-				Expect(err).ShouldNot(HaveOccurred())
-				result, err = client.List(metav1.ListOptions{})
-				Expect(err).ShouldNot(HaveOccurred())
-				Expect(result.Items).To(HaveLen(0))
-			})
-		})
-	})
-})
+	instance = Cluster{}
+	instance.Name = "instance-1"
+
+	expected = instance
+
+	// When sending a storage request for a valid config,
+	// it should provide CRUD access to the object.
+	client = cs.ClusterV1alpha1().Clusters("cluster-test-valid")
+
+	// Test that the create request returns success.
+	actual, err := client.Create(&instance)
+	assertNoError(t, err)
```

Comment:
When you use a helper function like this, you will see errors reported on the line where `t.Fatal` is called. That would be `v1alpha1_suite_test.go:32`. Also, if the error message isn't informative, then you have no idea where this error came from. This can be slightly alleviated by using `t.Helper()`, but it's often better to just write:

```go
if err != nil {
        t.Fatalf("Error creating cluster: %v", err)
}
```